<a href="https://colab.research.google.com/github/jittakorn-ch/data_viz_2022/blob/main/Project_Dataviz_2022.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import pandas as pd     # ใช้ทำงานกับ ตาราง

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
%cd /content/drive/MyDrive/Project_DataViz_2022

/content/drive/MyDrive/Project_DataViz_2022


# 1.covid data

## ตรวจสอบและรวมข้อมูล covid_data

- เนื่องจาก file covid_data มี 4 files จึงเช็คดูว่ามีข้อมูลซ้ำกันมั้ย

In [ ]:
covid_data1 = pd.read_csv('confirmed-cases-1.csv')
print(covid_data1.shape)
covid_data1.head()

(839771, 11)


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

In [ ]:
covid_data2 = pd.read_csv('confirmed-cases-since-120864.csv')
print(covid_data2.shape)
covid_data2.head()

(1042168, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN

In [ ]:
covid_data3 = pd.read_csv('confirmed-cases-since-271064.csv')
print(covid_data3.shape)
covid_data3.head()

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


(1010459, 12)


No. announce_date Notified date   sex   age Unit nationality  \
0  1859158    26/10/2021    25/10/2021   ชาย  20.0   ปี    Thailand   
1  1859159    26/10/2021    25/10/2021  หญิง  42.0   ปี    Thailand   
2  1859160    26/10/2021    25/10/2021  หญิง  33.0   ปี    Thailand   
3  1859161    26/10/2021    25/10/2021  หญิง  52.0   ปี    Thailand   
4  1859162    26/10/2021    25/10/2021  หญิง  84.0   ปี    Thailand   

  province_of_isolation                                           risk  \
0                 ลำปาง                                          อื่นๆ   
1                 ลำปาง                                          อื่นๆ   
2                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
3                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
4                 ลำปาง    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  Unnamed: 11  
0             ลำปาง        เมืองลำปาง          NaN  
1             ลำปาง        เมืองลำปาง          NaN  
2             ลำปาง          ห้างฉัตร          NaN  
3             ลำปาง          ห้างฉัตร          NaN  
4             ลำปาง        เมืองลำปาง          NaN

In [ ]:
covid_data4 = pd.read_csv('confirmed-cases-since-280265.csv')
print(covid_data4.shape)
covid_data4.head()

(433553, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  2869617     28/2/2022     27/2/2022   ชาย  41.0   ปี    Thailand   
1  2869618     28/2/2022     27/2/2022  หญิง  45.0   ปี    Thailand   
2  2869619     28/2/2022     27/2/2022  หญิง  40.0   ปี    Thailand   
3  2869620     28/2/2022     27/2/2022   ชาย  44.0   ปี    Thailand   
4  2869621     28/2/2022     27/2/2022   ชาย  11.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
1                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
2                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
3                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
4                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0             พะเยา        เมืองพะเยา  
1             พะเยา        เมืองพะเยา  
2             พะเยา        เมืองพะเยา  
3             พะเยา        เมืองพะเยา  
4             พะเยา        เมืองพะเยา

**ตรวจสอบว่าข้อมูลซ้ำกันหรือไม่**

In [ ]:
# หาค่า min max ของ column NO.
def min_max(i):
    a = min(i['No.'])
    b = max(i['No.'])
    print(f'min of NO. is {a}')
    print(f'max of NO. is {b}')

In [ ]:
min_max(covid_data1)    # ดูค่า min max ของตาราง covid_data1

min of NO. is 1
max of NO. is 839771


In [ ]:
min_max(covid_data2)    # ดูค่า min max ของตาราง covid_data2

min of NO. is 816990
max of NO. is 1859157


In [ ]:
min_max(covid_data3)    # ดูค่า min max ของตาราง covid_data3

min of NO. is 1859158
max of NO. is 2869616


In [ ]:
min_max(covid_data4)    # ดูค่า min max ของตาราง covid_data4

min of NO. is 2869617
max of NO. is 3303169


- เนื่องจาก ลักษณะของข้อมูลที่ดูจากค่าในคอลัมน์ NO. (รหัสผู้ป่วย) ตาราง covid_data1 และ covid_data2 จะมีค่าซ้ำกัน

(หรัสที่มากที่สุดใน covid_data1 > รหัสที่น้อยที่สุดใน covid_data2)

- ดูลักษณะข้อมูลว่าซ้ำกันจริงมั้ย

In [ ]:
covid_data1.iloc[816989:,:].head(10)    # ดูข้อมูล covid_data1 ตั้งแต่แถวที่ 816989 ถึงแถวสุดท้าย

No. announce_date Notified date   sex   age Unit nationality  \
816989  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
816990  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
816991  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
816992  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
816993  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
816994  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
816995  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
816996  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
816997  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
816998  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

       province_of_isolation                                         risk  \
816989              เชียงราย                                        อื่นๆ   
816990              เชียงราย                                        อื่นๆ   
816991              เชียงราย                                        อื่นๆ   
816992              เชียงราย                                        อื่นๆ   
816993              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
816994              เชียงราย                                        อื่นๆ   
816995              เชียงราย                                        อื่นๆ   
816996              เชียงราย                                        อื่นๆ   
816997              เชียงราย                                        อื่นๆ   
816998              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

       province_of_onset district_of_onset  
816989               NaN               NaN  
816990               NaN               NaN  
816991               NaN               NaN  
816992               NaN               NaN  
816993               NaN               NaN  
816994               NaN               NaN  
816995               NaN               NaN  
816996               NaN               NaN  
816997               NaN               NaN  
816998               NaN               NaN

In [ ]:
covid_data2.head(10)    # ดูข้อมูล covid_data2

No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
5  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
6  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
7  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
8  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
9  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
5              เชียงราย                                        อื่นๆ   
6              เชียงราย                                        อื่นๆ   
7              เชียงราย                                        อื่นๆ   
8              เชียงราย                                        อื่นๆ   
9              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  
5               NaN               NaN  
6               NaN               NaN  
7               NaN               NaN  
8               NaN               NaN  
9               NaN               NaN

- จากการดูลักษณะข้อมูล ข้อมูลซ้ำกันจริง

- ดังนั้น ลบข้อมูลที่ซ้ำออก โดยเลือกลบจากตาราง covid_data1

In [ ]:
covid_data1_drop = covid_data1.iloc[:816989,:]
covid_data1_drop.head()

No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

In [ ]:
min_max(covid_data1_drop)   # print ดู No. น้อยสุดและมากสุด ใน covid_data1_drop

min of NO. is 1
max of NO. is 816989


In [ ]:
min_max(covid_data4)     # print ดู No. น้อยสุดและมากสุด ใน covid_data4

min of NO. is 2869617
max of NO. is 3303169


In [ ]:
covid_data1_drop.shape[0] + covid_data2.shape[0] + covid_data3.shape[0] + covid_data4.shape[0]   # บวกดูจำนวนข้อมูลของทั้ง 4 ตาราง

3303169

- จะเห็นว่าจำนวน NO. ที่มากที่สุดของข้อมูล เท่ากับ จำนวนข้อมูลของทั้ง 4 ตาราง บวกกัน เท่ากับ 3303169 แสดงว่าข้อมูลแต่ละตารางไม่ได้เก็บซ้ำกันแล้ว

In [ ]:
# รวมข้อมูล covid_data
covid_data = pd.concat([covid_data1_drop, covid_data2, covid_data3, covid_data4])
print(covid_data.shape)
covid_data.head()

(3303169, 12)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  Unnamed: 11  
0               NaN          NaN  
1               NaN          NaN  
2             เมือง          NaN  
3               NaN          NaN  
4               NaN          NaN

In [ ]:
covid_df = covid_data[['No.','announce_date','sex','age','risk','province_of_onset','district_of_onset']]
print(covid_df.shape)
covid_df.head()

(3303169, 7)


No. announce_date   sex   age                              risk  \
0    1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร               NaN  
1     กรุงเทพมหานคร               NaN  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร               NaN  
4     กรุงเทพมหานคร               NaN

## Clean covid_data

In [ ]:
covid_df.isnull().sum()   # นับจำนวนค่า NaN ของแต่ละ columns

No.                       0
announce_date             0
sex                  179060
age                  266795
risk                   1187
province_of_onset    398224
district_of_onset    499368
dtype: int64

### C: district_of_onset

#### เพิ่ม กรุงเทพมหานคร ในอำเภอ

In [ ]:
new_district = covid_df[['province_of_onset','district_of_onset']].fillna(method='ffill',axis=1)    # เติมค่า nan ของคอลัมน์ district_of_onset ด้วยค่าในคอลัมน์ province_of_onset
print(new_district.shape)
new_district.head()

(3303169, 2)


province_of_onset district_of_onset
0     กรุงเทพมหานคร     กรุงเทพมหานคร
1     กรุงเทพมหานคร     กรุงเทพมหานคร
2            นครปฐม             เมือง
3     กรุงเทพมหานคร     กรุงเทพมหานคร
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
covid_df.loc[:,['district_of_onset']] = new_district.iloc[:,1]      # อัดเดท คอลัมน์ district_of_onset
print(covid_df.shape)
covid_df.head()

(3303169, 7)


/usr/local/lib/python3.7/dist-packages/pandas/core/indexing.py:1773: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  self._setitem_single_column(ilocs[0], value, pi)


No. announce_date   sex   age                              risk  \
0    1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

#### จัดรูปแบบข้อมูล

In [ ]:
len(set(covid_df['district_of_onset']))     # อำเภอ ในประเทศไทย มี 878 อำเภอ + 'กรุงเทพมหานคร' = 879 ข้อมูลต้องมีค่าไม่เกินนี้

6260

In [ ]:
district = ['กรุงเทพมหานคร','กงไกรลาศ',	'แกดำ',	'คีรีมาศ',	'เชียงกลาง',	'ตาพระยา',	'เทพา',	'เนินขาม',	'บางใหญ่',	'ปทุมรัตต์',	'พบพระ',	'ฟากท่า',	'ยะรัง',	'วังน้อย',	'ศรีวิไล',	'สามชุก',	'หนองสองห้อง',
            'กงหรา',	'แกลง',	'คีรีรัฐนิคม',	'เชียงขวัญ',	'ตาลสุม',	'เทพารักษ์',	'เนินมะปราง',	'บาเจาะ',	'ปทุมราชวงศา',	'พยัคฆภูมิพิสัย',	'ภักดีชุมพล',	'ยะหริ่ง',	'วังน้ำเขียว',	'ศรีสงคราม',	'สามพราน',	'หนองสูง',
            'กบินทร์บุรี',	'โกรกพระ',	'คุระบุรี',	'เชียงของ',	'เต่างอย',	'เทิง',	'เนินสง่า',	'บ้านกรวด',	'ประโคนชัย',	'พยุห์',	'ภาชี',	'ยะหา',	'วังน้ำเย็น',	'ศรีสมเด็จ',	'สามร้อยยอด',	'หนองเสือ',
            'กมลาไสย',	'โกสัมพีนคร',	'คูเมือง',	'เชียงคาน',	'ถลาง',	'ไทยเจริญ',	'โนนคูณ',	'บ้านเขว้า',	'ประจักษ์ศิลปาคม',	'พยุหะคีรี',	'ภูกระดึง',	'ยางชุมน้อย',	'วังโป่ง',	'ศรีสวัสดิ์',	'สายบุรี',	'หนองแสง',
            'กรงปินัง',	'โกสุมพิสัย',	'เคียนซา',	'เชียงคำ',	'ถ้ำพรรณรา',	'ไทรงาม',	'โนนดินแดง',	'บ้านคา',	'ประจันตคาม',	'พรเจริญ',	'ภูกามยาว',	'ยางตลาด',	'วังม่วง',	'ศรีสัชนาลัย',	'สารภี',	'หนองหงส์',
            'กระทุ่มแบน',	'ขนอม',	'แคนดง',	'เชียงดาว',	'เถิน',	'ไทรน้อย',	'โนนแดง',	'บ้านค่าย',	'ประทาย',	'พรรณนานิคม',	'ภูเขียว',	'ยางสีสุราช',	'วังยาง',	'ศรีสาคร',	'สำโรง',	'หนองหญ้าไซ',
            'กระนวน',	'ขลุง',	'โคกเจริญ',	'เชียงม่วน',	'ทรายทองวัฒนา',	'ไทรโยค',	'โนนไทย',	'บ้านโคก',	'ปรางค์กู่',	'พรหมคีรี',	'ภูซาง',	'ย่านตาขาว',	'วังวิเศษ',	'ศรีสำโรง',	'สำโรงทาบ',	'หนองหญ้าปล้อง',
            'กระบุรี',	'ขาณุวรลักษบุรี',	'โคกโพธิ์',	'เชียงยืน',	'ทรายมูล',	'ธวัชบุรี',	'โนนนารายณ์',	'บ้านฉาง',	'ปราณบุรี',	'พรหมบุรี',	'ภูผาม่าน',	'ยี่งอ',	'วังสมบูรณ์',	'ศิลาลาด',	'สิเกา',	'หนองหาน',
            'กระสัง',	'ขามทะเลสอ',	'โคกโพธิ์ไชย',	'เชียงแสน',	'ทองผาภูมิ',	'ธัญบุรี',	'โนนศิลา',	'บ้านด่าน',	'ปราสาท',	'พรหมพิราม',	'ภูพาน',	'ร้องกวาง',	'วังสะพุง',	'ศีขรภูมิ',	'สิงหนคร',	'หนองหิน',
            'กระแสสินธุ์',	'ขามสะแกแสง',	'โคกศรีสุพรรณ',	'เชียรใหญ่',	'ทองแสนขัน',	'ธาตุพนม',	'โนนสะอาด',	'บ้านด่านลานหอย',	'ปลวกแดง',	'พระทองคำ',	'ภูเพียง',	'ร่องคำ',	'วังสามหมอ',	'สตึก',	'สิชล',	'หนองใหญ่',
            'กะทู้',	'ขุขันธ์',	'โคกสำโรง',	'โชคชัย',	'ทับคล้อ',	'ธารโต',	'โนนสัง',	'บ้านดุง',	'ปลาปาก',	'พระนครศรีอยุธยา',	'ภูเรือ',	'ร่อนพิบูลย์',	'วังหิน',	'สทิงพระ',	'สิรินธร',	'หนองฮี',
            'กะปง',	'ขุนตาล',	'โคกสูง',	'ไชยปราการ',	'ทับปุด',	'นครชัยศรี',	'โนนสุวรรณ',	'บ้านตาก',	'ปลายพระยา',	'พระประแดง',	'ภูเวียง',	'ระแงะ',	'วังเหนือ',	'สนม',	'สีคิ้ว',	'หล่มเก่า',
            'กะเปอร์',	'ขุนยวม',	'ฆ้องชัย',	'ไชยวาน',	'ทับสะแก',	'นครไทย',	'โนนสูง',	'บ้านตาขุน',	'ปะคำ',	'พระพรหม',	'ภูสิงห์',	'ระโนด',	'วัฒนานคร',	'สนามชัยเขต',	'สีชมพู',	'หล่มสัก',
            'กะพ้อ',	'ขุนหาญ',	'งาว',	'ไชยา',	'ทัพทัน',	'นครหลวง',	'บรบือ',	'บ้านแท่น',	'ปะทิว',	'พระพุทธบาท',	'ภูหลวง',	'รัตนบุรี',	'วัดโบสถ์',	'สบปราบ',	'สีดา',	'หลังสวน',
            'กันตัง',	'เขมราฐ',	'จตุรพักตรพิมาน',	'ไชโย',	'ท่าคันโท',	'นบพิตำ',	'บรรพตพิสัย',	'บ้านธิ',	'ปะนาเระ',	'พระยืน',	'มโนรมย์',	'รัตนวาปี',	'วัดเพลง',	'สบเมย',	'สุขสำราญ',	'ห้วยกระเจา',
            'กันทรลักษ์',	'เขวาสินรินทร์',	'จอมทอง',	'ซับใหญ่',	'ท่าฉาง',	'นากลาง',	'บ่อเกลือ',	'บ้านนา',	'ปะเหลียน',	'พระสมุทรเจดีย์',	'มวกเหล็ก',	'รัตภูมิ',	'วัดสิงห์',	'สมเด็จ',	'สุคิริน',	'ห้วยคต',
            'กันทรวิชัย',	'เขาค้อ',	'จอมบึง',	'ซำสูง',	'ท่าชนะ',	'นาแก',	'บ่อทอง',	'บ้านนาเดิม',	'ปักธงชัย',	'พระแสง',	'ม่วงสามสิบ',	'รัษฎา',	'วานรนิวาส',	'สรรคบุรี',	'สุวรรณคูหา',	'ห้วยแถลง',
            'กันทรารมย์',	'เขาคิชฌกูฏ',	'จอมพระ',	'เซกา',	'ท่าช้าง',	'นาคู',	'บ่อพลอย',	'บ้านนาสาร',	'ปัว',	'พรานกระต่าย',	'มหาชนะชัย',	'ราชสาส์น',	'วาปีปทุม',	'สรรพยา',	'สุวรรณภูมิ',	'ห้วยทับทัน',
            'กัลยาณิวัฒนา',	'เขาฉกรรจ์',	'จะนะ',	'โซ่พิสัย',	'ท่าแซะ',	'นางรอง',	'บ่อไร่',	'บ้านบึง',	'ปากเกร็ด',	'พร้าว',	'มหาราช',	'รามัน',	'วาริชภูมิ',	'สระใคร',	'สุไหงโก-ลก',	'ห้วยผึ้ง',
            'กาญจนดิษฐ์',	'เขาชะเมา',	'จะแนะ',	'ดงเจริญ',	'ท่าตะเกียบ',	'นาจะหลวย',	'บันนังสตา',	'บ้านโป่ง',	'ปากคาด',	'พล',	'มะขาม',	'ราษีไศล',	'วารินชำราบ',	'สระโบสถ์',	'สุไหงปาดี',	'ห้วยเม็ก',
            'กาบเชิง',	'เขาชัยสน',	'จักราช',	'ดงหลวง',	'ท่าตะโก',	'นาเชือก',	'บัวเชด',	'บ้านผือ',	'ปากชม',	'พลับพลาชัย',	'มะนัง',	'รือเสาะ',	'วิเชียรบุรี',	'สร้างคอม',	'สูงเนิน',	'ห้วยยอด',
            'กาบัง',	'เขาพนม',	'จังหาร',	'ดอกคำใต้',	'ท่าตูม',	'นาด้วง',	'บัวลาย',	'บ้านไผ่',	'ปากช่อง',	'พะโต๊ะ',	'มัญจาคีรี',	'เรณูนคร',	'วิภาวดี',	'สวนผึ้ง',	'สูงเม่น',	'ห้วยราช',
            'กำแพงแสน',	'เขาย้อย',	'จัตุรัส',	'ดอนจาน',	'ท่าบ่อ',	'นาดี',	'บัวใหญ่',	'บ้านฝาง',	'ปากท่อ',	'พังโคน',	'มายอ',	'ลอง',	'วิเศษชัยชาญ',	'สวรรคโลก',	'เสนา',	'หว้านใหญ่',
            'กุฉินารายณ์',	'เขาวง',	'จุน',	'ดอนเจดีย์',	'ท่าปลา',	'นาดูน',	'บางกรวย',	'บ้านแพง',	'ปากพนัง',	'พัฒนานิคม',	'เมยวดี',	'ละงู',	'วิหารแดง',	'สว่างแดนดิน',	'เสนางคนิคม',	'หันคา',
            'กุดข้าวปุ้น',	'เขาสมิง',	'จุฬาภรณ์',	'ดอนตาล',	'ท่าแพ',	'นาตาล',	'บางกระทุ่ม',	'บ้านแพรก',	'ปากพลี',	'พาน',	'เมือง',	'ละแม',	'เวียงเก่า',	'สว่างวีระวงศ์',	'เสริมงาม',	'หัวตะพาน',
            'กุดจับ',	'เขาสวนกวาง',	'เจริญศิลป์',	'ดอนตูม',	'ท่าม่วง',	'นาทม',	'บางกล่ำ',	'บ้านแพ้ว',	'ปากพะยูน',	'พานทอง',	'แม่จริม',	'ละหานทราย',	'เวียงแก่น',	'สว่างอารมณ์',	'เสลภูมิ',	'หัวไทร',
            'กุดชุม',	'เขื่องใน',	'เจาะไอร้อง',	'ดอนพุด',	'ท่ามะกา',	'นาทวี',	'บางแก้ว',	'บ้านโพธิ์',	'ปางมะผ้า',	'พิชัย',	'แม่จัน',	'ละอุ่น',	'เวียงชัย',	'สวี',	'เสาไห้',	'หัวหิน',
            'กุดบาก',	'โขงเจียม',	'แจ้ห่ม',	'ดอนมดแดง',	'ท้ายเหมือง',	'นาน้อย',	'บางขัน',	'บ้านม่วง',	'ปางศิลาทอง',	'พิบูลมังสาหาร',	'แม่แจ่ม',	'ลับแล',	'เวียงเชียงรุ้ง',	'สหัสขันธ์',	'เสิงสาง',	'ห้างฉัตร',
            'กุดรัง',	'คง',	'ฉวาง',	'ดอนสัก',	'ท่ายาง',	'นาบอน',	'บางคนที',	'บ้านไร่',	'ป่าซาง',	'พิบูลย์รักษ์',	'แม่ใจ',	'ลาดบัวหลวง',	'เวียงป่าเป้า',	'สอง',	'แสวงหา',	'หางดง',
            'กุมภวาปี',	'ครบุรี',	'เฉลิมพระเกียรติ',	'ดอยเต่า',	'ท่าเรือ',	'นาโพธิ์',	'บางคล้า',	'บ้านลาด',	'ป่าแดด',	'พิปูน',	'แม่แตง',	'ลาดยาว',	'เวียงสระ',	'สองแคว',	'หนองกี่',	'หาดสำราญ',
            'กุยบุรี',	'คลองขลุง',	'เฉลิมพระเกียรติ',	'ดอยสะเก็ด',	'ท่าลี่',	'นามน',	'บางซ้าย',	'บ้านสร้าง',	'ป่าติ้ว',	'พิมาย',	'แม่ทะ',	'ลาดหลุมแก้ว',	'เวียงสา',	'ส่องดาว',	'หนองกุงศรี',	'หาดใหญ่',
            'กุสุมาลย์',	'คลองเขื่อน',	'เฉลิมพระเกียรติ',	'ดอยหลวง',	'ท่าวังผา',	'นายายอาม',	'บางไทร',	'บ้านหมอ',	'ป่าบอน',	'พุทธมณฑล',	'แม่ทา',	'ลานกระบือ',	'เวียงหนองล่อง',	'สองพี่น้อง',	'หนองขาหย่าง',	'เหนือคลอง',
            'กู่แก้ว',	'คลองท่อม',	'เฉลิมพระเกียรติ',	'ดอยหล่อ',	'ท่าวุ้ง',	'นายูง',	'บางน้ำเปรี้ยว',	'บ้านหมี่',	'ป่าพะยอม',	'พุทไธสง',	'แม่เปิน',	'ลานสกา',	'เวียงแหง',	'สอยดาว',	'หนองแค',	'เหล่าเสือโก้ก',
            'เกษตรวิสัย',	'คลองลาน',	'เฉลิมพระเกียรติ',	'ด่านขุนทด',	'ท่าศาลา',	'นาเยีย',	'บางบ่อ',	'บ้านหลวง',	'ป่าโมก',	'พุนพิน',	'แม่พริก',	'ลานสัก',	'แว้ง',	'สะเดา',	'หนองจิก',	'แหลมงอบ',
            'เกษตรสมบูรณ์',	'คลองหลวง',	'ชนแดน',	'ด่านช้าง',	'ท่าสองยาง',	'นาโยง',	'บางบัวทอง',	'บ้านเหลื่อม',	'ปาย',	'เพ็ญ',	'แม่ฟ้าหลวง',	'ลำดวน',	'แวงน้อย',	'สะบ้าย้อย',	'หนองฉาง',	'แหลมสิงห์',
            'เก้าเลี้ยว',	'คลองหอยโข่ง',	'ชนบท',	'ด่านซ้าย',	'ท่าหลวง',	'นาวัง',	'บางบาล',	'บ้านแหลม',	'เปือยน้อย',	'โพทะเล',	'แม่เมาะ',	'ลำทะเมนชัย',	'แวงใหญ่',	'สะเมิง',	'หนองแซง',	'องครักษ์',
            'เกาะกูด',	'คลองหาด',	'ชะอวด',	'ด่านมะขามเตี้ย',	'ท่าใหม่',	'นาหม่อม',	'บางปลาม้า',	'บ้านใหม่ไชยพจน์',	'แปลงยาว',	'โพธาราม',	'แม่ระมาด',	'ลำทับ',	'ศรีเชียงใหม่',	'สังขละบุรี',	'หนองโดน',	'อมก๋อย',
            'เกาะคา',	'คลองใหญ่',	'ชะอำ',	'ดำเนินสะดวก',	'ท่าอุเทน',	'นาหมื่น',	'บางปะกง',	'บ้านแฮด',	'โป่งน้ำร้อน',	'โพธิ์ชัย',	'แม่ริม',	'ลำปลายมาศ',	'ศรีณรงค์',	'สังขะ',	'หนองนาคำ',	'อรัญประเทศ',
            'เกาะจันทร์',	'ควนกาหลง',	'ชัยบาดาล',	'เดชอุดม',	'ทุ่งเขาหลวง',	'นาหว้า',	'บางปะหัน',	'บ้านโฮ่ง',	'ผักไห่',	'โพธิ์ตาก',	'แม่ลาน',	'ลำลูกกา',	'ศรีเทพ',	'สังคม',	'หนองบัว',	'อัมพวา',
            'เกาะช้าง',	'ควนขนุน',	'ชัยบุรี',	'เด่นชัย',	'ทุ่งช้าง',	'นาแห้ว',	'บางปะอิน',	'บำเหน็จณรงค์',	'ผาขาว',	'โพธิ์ทอง',	'แม่ลาน้อย',	'ลำสนธิ',	'ศรีธาตุ',	'สัตหีบ',	'หนองบัวแดง',	'อากาศอำนวย',
            'เกาะพะงัน',	'ควนโดน',	'ช้างกลาง',	'เดิมบางนางบวช',	'ทุ่งตะโก',	'น้ำเกลี้ยง',	'บางพลี',	'บึงโขงหลง',	'ฝาง',	'โพธิ์ไทร',	'แม่ลาว',	'ลี้',	'ศรีนคร',	'สันกำแพง',	'หนองบัวระเหว',	'อาจสามารถ',
            'เกาะยาว',	'ควนเนียง',	'ชาติตระการ',	'ตรอน',	'ทุ่งฝน',	'น้ำขุ่น',	'บางแพ',	'บึงนาราง',	'เฝ้าไร่',	'โพธิ์ประทับช้าง',	'แม่วงก์',	'ลืออำนาจ',	'ศรีนครินทร์',	'สันติสุข',	'หนองบุญมาก',	'อ่าวลึก',
            'เกาะลันตา',	'คอนสวรรค์',	'ชานุมาน',	'ตระการพืชผล',	'ทุ่งยางแดง',	'น้ำปาด',	'บางมูลนาก',	'บึงบูรพ์',	'พญาเม็งราย',	'โพธิ์ศรีสุวรรณ',	'แม่วาง',	'เลาขวัญ',	'ศรีบรรพต',	'สันทราย',	'หนองปรือ',	'อินทร์บุรี',
            'เกาะสมุย',	'คอนสาร',	'ชำนิ',	'ตะกั่วทุ่ง',	'ทุ่งศรีอุดม',	'น้ำพอง',	'บางระกำ',	'บึงสามพัน',	'พนม',	'โพนทราย',	'แม่สรวย',	'เลิงนกทา',	'ศรีบุญเรือง',	'สันป่าตอง',	'หนองไผ่',	'อุทัย',
            'เกาะสีชัง',	'ค้อวัง',	'ชื่นชม',	'ตะกั่วป่า',	'ทุ่งสง',	'น้ำยืน',	'บางระจัน',	'บึงสามัคคี',	'พนมดงรัก',	'โพนทอง',	'แม่สอด',	'วชิรบารมี',	'ศรีประจันต์',	'สากเหล็ก',	'หนองพอก',	'อุทุมพรพิสัย',
            'แก่งกระจาน',	'ค่ายบางระจัน',	'ชุมตาบง',	'ตะพานหิน',	'ทุ่งเสลี่ยม',	'น้ำโสม',	'บางละมุง',	'บุ่งคล้า',	'พนมทวน',	'โพนนาแก้ว',	'แม่สะเรียง',	'วังจันทร์',	'ศรีมหาโพธิ',	'สามโก้',	'หนองม่วง',	'อุบลรัตน์',
            'แก้งคร้อ',	'คำเขื่อนแก้ว',	'ชุมพลบุรี',	'ตะโหมด',	'ทุ่งหว้า',	'น้ำหนาว',	'บางเลน',	'บุณฑริก',	'พนมไพร',	'โพนพิสัย',	'แม่สาย',	'วังเจ้า',	'ศรีมโหสถ',	'สามโคก',	'หนองม่วงไข่',	'อุ้มผาง',
            'แก่งคอย',	'คำชะอี',	'ชุมพวง',	'ตากใบ',	'ทุ่งหัวช้าง',	'นิคมคำสร้อย',	'บางสะพาน',	'เบญจลักษ์',	'พนมสารคาม',	'โพนสวรรค์',	'แม่ออน',	'วังชิ้น',	'ศรีเมืองใหม่',	'สามง่าม',	'หนองมะโมง',	'อู่ทอง',
            'แก้งสนามนาง',	'คำตากล้า',	'ชุมแพ',	'ตากฟ้า',	'ทุ่งใหญ่',	'นิคมน้ำอูน',	'บางสะพานน้อย',	'เบตง',	'พนัสนิคม',	'ไพรบึง',	'แม่อาย',	'วังทรายพูน',	'ศรีรัตนะ',	'สามเงา',	'หนองเรือ',	'เอราวัณ',
            'แก่งหางแมว',	'คำม่วง',	'ชุมแสง',	'ตาคลี',	'เทพสถิต',	'นิคมพัฒนา',	'บางเสาธง',	'ปง',	'พนา',	'ไพศาลี',	'ไม้แก่น',	'วังทอง',	'ศรีราชา',	'สามชัย',	'หนองวัวซอ',	'ฮอด']


In [ ]:
district_c = covid_df[covid_df['district_of_onset'].isin(district)]
district_c

No. announce_date   sex   age                              risk  \
0             1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1             2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2             3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3             4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4             5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
...         ...           ...   ...   ...                               ...   
433438  3303055     18/3/2022   NaN  60.0                             อื่นๆ   
433439  3303056     18/3/2022   NaN  57.0                             อื่นๆ   
433440  3303057     18/3/2022   NaN  31.0                             อื่นๆ   
433444  3303061     18/3/2022   NaN  11.0                             อื่นๆ   
433449  3303066     18/3/2022   NaN  64.0                             อื่นๆ   

       province_of_onset district_of_onset  
0          กรุงเทพมหานคร     กรุงเทพมหานคร  
1          กรุงเทพมหานคร     กรุงเทพมหานคร  
2                 นครปฐม             เมือง  
3          กรุงเทพมหานคร     กรุงเทพมหานคร  
4          กรุงเทพมหานคร     กรุงเทพมหานคร  
...                  ...               ...  
433438            นครปฐม           สามพราน  
433439            นครปฐม           สามพราน  
433440          ศรีสะเกษ           ภูสิงห์  
433444        นครราชสีมา        หนองบุญมาก  
433449         กำแพงเพชร          คลองขลุง  

[1805800 rows x 7 columns]

In [ ]:
len(set(district_c['district_of_onset']))     # 878 อำเภอ

797

In [ ]:
district_c.isnull().sum()

No.                      0
announce_date            0
sex                  61178
age                  91882
risk                   608
province_of_onset    70721
district_of_onset        0
dtype: int64

### C: province_of_onset

In [ ]:
df_drop = district_c.dropna(subset=['province_of_onset'])       # ลบแถวที่ คอลัมน์ province_of_onset หรือ district_of_onset เป็นค่า nan
print(df_drop.shape)
df_drop.head()

(1735079, 7)


No. announce_date   sex   age                              risk  \
0    1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
df_drop.isnull().sum()

No.                      0
announce_date            0
sex                  49933
age                  77964
risk                   498
province_of_onset        0
district_of_onset        0
dtype: int64

#### จัดรูปแบบข้อมูล

In [ ]:
len(set(df_drop['province_of_onset']))  #ข้อมูลในคอลัมน์ province_of_onset ต้องมี 77 ค่า เพราะมี 77 จังหวัด ถ้ามากว่า 77 อาจเป็นเพราะคีร์ข้อมูลไม่ตรงกัน

331

In [ ]:
# เปลี่ยนข้อมูลให้อยู่ในรูปแบบเดียวกัน
df_drop['province_of_onset'].replace({'\xa0กาญจนบุรี':'กาญจนบุรี','\xa0ชลบุรี':'ชลบุรี','\xa0ตราด':'ตราด','\xa0นครปฐม':'นครปฐม',
                                         '\xa0สมุทรสงคราม':'สมุทรสงคราม','\xa0อุทัยธานี':'อุทัยธานี','กทม':'กรุงเทพมหานคร','กาฬสินธิ์':'กาฬสินธุ์',
                                         'กาฬสินธ์':'กาฬสินธุ์','ขอนแก่่น':'ขอนแก่น','ชลบุรี\t':'ชลบุรี','ชุุมพร':'ชุมพร','ท้ายบ้าน':'สมุทรปราการ',
                                         'นครปบม':'นครปฐม','ธารโต':'ยะลา','นครศร๊ะรรมราช':'นครศรีธรรมราช','นตรปฐม':'นครปฐม',
                                         'นาราธิวาส':'นราธิวาส','นาวัง':'หนองบัวลำภู','บางกะปิ':'กรุงเทพมหานคร','บึงโขงหลง':'บึงกาฬ',
                                         'บุรีรัมย์\xa0':'บุรีรัมย์','บุรีรัยม์':'บุรีรัมย์','บ้านหลวง':'น่าน','ปทุมธานี.':'ปทุมธานี','ปทุมธานีี':'ปทุมธานี',
                                         'ปทุุมธานี':'ปทุมธานี','ปมุมธานี':'ปทุมธานี','ประจวบฯ':'ประจวบคีรีขันธ์','ปลวกแดง':'ระยอง','ปลายพระยา':'กระบี่',
                                         'ปัว':'น่าน','ปุทมธานี':'ปทุมธานี','พิษณูโลก':'พิษณุโลก','ภูเพียง':'น่าน','มหาสาคาม':'มหาสารคาม',
                                         'ยานนาวา':'กรุงเทพมหานคร','ร้องกวาง':'แพร่','ลอง':'แพร่','ลาดพร้าว':'กรุงเทพมหานคร','ลำลูกกา':'ปทุมธานี',
                                         'วังชิ้น':'ลำปาง','ศรีษะเกษ':'ศรีสะเกษ','สงขลา\u200b':'สงขลา','สมถทรปราการ':'สมุทรปราการ','สมทรสาคร':'สมุทรสาคร',
                                         'สมุครปราการ':'สมุทรปราการ','สมุทรปรากร':'สมุทรปราการ','สมุทรปรากรร':'สมุทรปราการ','สมุทราปราการ':'สมุทรปราการ','สมุทสงคราม':'สมุทรสงคราม',
                                         'สมุทสาคร':'สมุทรสาคร','สมุุทรปราการ':'สมุทรปราการ','สมุุทรสาคร':'สมุทรสาคร','สอง':'แพร่','สองแคว':'น่าน',
                                         'สะแก้ว':'สระแก้ว','สารภี':'เชียงใหม่','สุทรปราการ':'สมุทรปราการ','สุมทรสาคร':'สมุทรสาคร','สุราษฎร์์ธานี':'สุราษฎร์ธานี',
                                         'สุราษฏร์ธานี':'สุราษฎร์ธานี','สุุราษฎร์ธานี':'สุราษฎร์ธานี','สูงเม่น':'แพร่','หนองบััวลำภู':'หนองบัวลำภู','หนองม่วงไข่':'แพร่',
                                         'อยุทธยา':'พระนครศรีอยุธยา','อยุธยา':'พระนครศรีอยุธยา','อยุุธยา':'พระนครศรีอยุธยา','อุดรธานีี':'อุดรธานี','อุบลราขธานี':'อุบลราชธานี',
                                         'อุุดรธาน':'อุดรธานี','อุุดรธานี':'อุดรธานี','อุุตรดิตถ์':'อุตรดิตถ์','อุุทัยธานี':'อุทัยธานี','เชีียงราย':'เชียงราย','เด่นชัย':'แพร่',
                                         'เพชรบุูรณ์':'เพชรบูรณ์','เมือง':'แพร่','เมืองน่าน':'น่าน','เรือ':'เลย','เวียงสา':'น่าน','\xa0สมุทรสาคร':'สมุทรสาคร',
                                         'สุมทรปราการ':'สมุทรปราการ','สุมุทรปราการ':'สมุทรปราการ','สุมุทรสาคร':'สมุทรสาคร','ชลบุุรี':'ชลบุรี','นคนปฐม':'นครปฐม',
                                         'นคปฐม':'นครปฐม','ขัยนาท':'ชัยนาท','ชมพร':'ชุมพร','นครนาชสีมา':'นครราชสีมา','สุวรรณภูมิ':'ร้อยเอ็ด',
                                         'สถานที่ทำงาน':'สงขลา','นอกเขต':'พิษณุโลก','แล้วป่วย':'อุบลราชธานี'}, inplace=True)

/usr/local/lib/python3.7/dist-packages/pandas/core/generic.py:6619: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return self._update_inplace(result)


In [ ]:
# เปลี่ยนทีละตัว ช้าและยุ่งยากเกินไป
len(set(df_drop['province_of_onset']))

267

In [ ]:
city = ['อำนาจเจริญ','บึงกาฬ', 'บุรีรัมย์', 'ชัยภูมิ', 'กาฬสินธุ์', 'ขอนแก่น', 'เลย', 'มหาสารคาม', 'มุกดาหาร', 'นครพนม', 'นครราชสีมา', 'หนองบัวลำภู', 'หนองคาย',
        'ร้อยเอ็ด', 'สกลนคร', 'ศรีสะเกษ', 'สุรินทร์', 'อุบลราชธานี', 'อุดรธานี', 'ยโสธร',
        'เชียงใหม่', 'เชียงราย', 'ลำปาง', 'ลำพูน', 'แม่ฮ่องสอน', 'น่าน', 'พะเยา', 'แพร่', 'อุตรดิตถ์','ตาก','สุโขทัย', 'พิษณุโลก', 'พิจิตร', 'กำแพงเพชร', 'เพชรบูรณ์',
        'นครสวรรค์', 'อุทัยธานี','อ่างทอง', 'ชัยนาท', 'พระนครศรีอยุธยา', 'กรุงเทพมหานคร','ลพบุรี', 'นครปฐม', 'นนทบุรี', 'ปทุมธานี', 'สมุทรปราการ', 'สมุทรสาคร', 'สมุทรสงคราม',
        'สระบุรี', 'สิงห์บุรี', 'สุพรรณบุรี','นครนายก','ฉะเชิงเทรา', 'จันทบุรี', 'ชลบุรี', 'ปราจีนบุรี', 'ระยอง', 'สระแก้ว', 'ตราด','กาญจนบุรี', 'ราชบุรี','เพชรบุรี', 'ประจวบคีรีขันธ์',
        'ชุมพร', 'นครศรีธรรมราช','นราธิวาส', 'ปัตตานี', 'พัทลุง', 'สงขลา', 'สุราษฎร์ธานี', 'ยะลา','กระบี่', 'พังงา', 'ภูเก็ต', 'ระนอง', 'สตูล', 'ตรัง']

In [ ]:
df_drop = df_drop[df_drop['province_of_onset'].isin(city)]    # .isin คือ เลือกข้อมูลที่ต้องการ
print(df_drop.shape)
df_drop.head()

(1734198, 7)


No. announce_date   sex   age                              risk  \
0    1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
len(set(df_drop['province_of_onset']))

77

### C: age

In [ ]:
df_drop.dtypes

No.                   int64
announce_date        object
sex                  object
age                  object
risk                 object
province_of_onset    object
district_of_onset    object
dtype: object

In [ ]:
df_drop['age'] = pd.to_numeric(df_drop['age'], errors='coerce')     # เปลี่ยน tpye ข้อมูล คอลัมน์ age
df_drop.dtypes

No.                    int64
announce_date         object
sex                   object
age                  float64
risk                  object
province_of_onset     object
district_of_onset     object
dtype: object

In [ ]:
mean_age = df_drop['age'].mean()
mean_age

35.00013781914152

In [ ]:
df_drop[['age']] = df_drop[['age']].fillna(mean_age)     # แทนค่า nan ของ age ด้วย mean

In [ ]:
df_drop.isnull().sum()

No.                      0
announce_date            0
sex                  49927
age                      0
risk                   498
province_of_onset        0
district_of_onset        0
dtype: int64

### C: sex & risk

In [ ]:
df_drop = df_drop.fillna('ไม่ระบุ')

In [ ]:
df_drop.isnull().sum()

No.                  0
announce_date        0
sex                  0
age                  0
risk                 0
province_of_onset    0
district_of_onset    0
dtype: int64

#### จัดรูปแบบข้อมูล

In [ ]:
set(df_drop['sex'])

{'ช', 'ชาย', 'ญ', 'นาย', 'หญิง', 'หณิง', 'หห', 'ไม่ระบุ'}

In [ ]:
df_drop.replace({'ช':'ชาย',
                 'นาย':'ชาย',
                 'ร.ต.ท.':'ชาย',
                 'ญ':'หญิง',
                 'หณิง':'หญิง',
                 'หห':'หญิง'},inplace=True)

In [ ]:
set(df_drop['sex'])

{'ชาย', 'หญิง', 'ไม่ระบุ'}

### C: announce_date

In [ ]:
print(df_drop.shape)
print(df_drop.dtypes)
df_drop.head()

(1734198, 7)
No.                    int64
announce_date         object
sex                   object
age                  float64
risk                  object
province_of_onset     object
district_of_onset     object
dtype: object


No. announce_date   sex   age                              risk  \
0    1     12/1/2020  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2     17/1/2020  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3     22/1/2020  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4     22/1/2020   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5     24/1/2020  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
df_drop['announce_date'] = pd.to_datetime(df_drop['announce_date'],format="%d/%m/%Y")

In [ ]:
print(df_drop.shape)
print(df_drop.dtypes)
df_drop.head()

(1734198, 7)
No.                           int64
announce_date        datetime64[ns]
sex                          object
age                         float64
risk                         object
province_of_onset            object
district_of_onset            object
dtype: object


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4    2020-01-22   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5    2020-01-24  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
before = covid_data.shape[0]
after = df_drop.shape[0]
lose = (before - after)*100/before

print(f'ข้อมูลเริ่ม้นเท่ากับ                  {before} แถว')
print(f'ข้อมูลหลังทำความสะอาดเหลือเท่ากับ   {after} แถว')
print(f'ข้อมูลหายไปทั้งหมด                {lose} %')

ข้อมูลเริ่ม้นเท่ากับ                  3303169 แถว
ข้อมูลหลังทำความสะอาดเหลือเท่ากับ   1734198 แถว
ข้อมูลหายไปทั้งหมด                47.498962360085116 %


In [ ]:
#####################################################################################################################################################

In [ ]:
print(df_drop.shape)    ############## ตารางสุดท้าย พร้อมใช้งานของ covid_data
df_drop.head()

(1734198, 7)


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
3    4    2020-01-22   ชาย  68.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
4    5    2020-01-24  หญิง  66.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง  
3     กรุงเทพมหานคร     กรุงเทพมหานคร  
4     กรุงเทพมหานคร     กรุงเทพมหานคร

In [ ]:
# df_drop.to_csv('covid.csv',index=False)     # โหลดตาราง

In [ ]:
# df_drop.to_csv('covid_cleaned.csv')     # โหลดตารางลง drive

# 2.region data

In [ ]:
region1 = { 'จังหวัด': ['อำนาจเจริญ','บึงกาฬ', 'บุรีรัมย์', 'ชัยภูมิ', 'กาฬสินธุ์', 'ขอนแก่น', 'เลย', 'มหาสารคาม', 'มุกดาหาร', 'นครพนม', 'นครราชสีมา', 'หนองบัวลำภู', 'หนองคาย', 'ร้อยเอ็ด', 'สกลนคร', 'ศรีสะเกษ', 'สุรินทร์', 'อุบลราชธานี', 'อุดรธานี', 'ยโสธร'],
             'ภูมิศาสตร์': ['ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ'],
             'การเมือง': ['ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ'],
             'อุตุนิยมวิทยา': ['ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ'],
             'การท่องเที่ยว': ['ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ', 'ตะวันออกเฉียงเหนือ']
}
region2 = { 'จังหวัด': ['เชียงใหม่', 'เชียงราย', 'ลำปาง', 'ลำพูน', 'แม่ฮ่องสอน', 'น่าน', 'พะเยา', 'แพร่', 'อุตรดิตถ์'],
             'ภูมิศาสตร์': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ'],
             'การเมือง': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ'],
             'อุตุนิยมวิทยา': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ'],
             'การท่องเที่ยว': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ']
}
region3 = { 'จังหวัด': ['ตาก'],
             'ภูมิศาสตร์': ['ตะวันตก'],
             'การเมือง': ['เหนือ'],
             'อุตุนิยมวิทยา': ['เหนือ'],
             'การท่องเที่ยว': ['เหนือ']
}
region4 = { 'จังหวัด': ['สุโขทัย', 'พิษณุโลก', 'พิจิตร', 'กำแพงเพชร', 'เพชรบูรณ์'],
             'ภูมิศาสตร์': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง'],
             'การเมือง': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ'],
             'อุตุนิยมวิทยา': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ'],
             'การท่องเที่ยว': ['เหนือ', 'เหนือ', 'เหนือ', 'เหนือ', 'เหนือ']
}
region5 = { 'จังหวัด': ['นครสวรรค์', 'อุทัยธานี'],
             'ภูมิศาสตร์': ['กลาง','กลาง'],
             'การเมือง': ['เหนือ','เหนือ'],
             'อุตุนิยมวิทยา': ['กลาง','กลาง'],
             'การท่องเที่ยว': ['เหนือ','เหนือ']
}
region6 = { 'จังหวัด': ['อ่างทอง', 'ชัยนาท', 'พระนครศรีอยุธยา', 'กรุงเทพมหานคร','ลพบุรี', 'นครปฐม', 'นนทบุรี', 'ปทุมธานี', 'สมุทรปราการ', 'สมุทรสาคร', 'สมุทรสงคราม', 'สระบุรี', 'สิงห์บุรี', 'สุพรรณบุรี'],
             'ภูมิศาสตร์': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง'],
             'การเมือง': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง'],
             'อุตุนิยมวิทยา': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง'],
             'การท่องเที่ยว': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง']
}
region7 = { 'จังหวัด': ['นครนายก'],
             'ภูมิศาสตร์': ['กลาง'],
             'การเมือง': ['กลาง'],
             'อุตุนิยมวิทยา': ['ตะวันออก'],
             'การท่องเที่ยว': ['กลาง']
}
region8 = { 'จังหวัด': ['ฉะเชิงเทรา', 'จันทบุรี', 'ชลบุรี', 'ปราจีนบุรี', 'ระยอง', 'สระแก้ว', 'ตราด'],
             'ภูมิศาสตร์': ['ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก'],
             'การเมือง': ['กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง', 'กลาง'],
             'อุตุนิยมวิทยา': ['ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก'],
             'การท่องเที่ยว': ['ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก', 'ตะวันออก']
}
region9 = { 'จังหวัด': ['กาญจนบุรี', 'ราชบุรี'],
             'ภูมิศาสตร์': ['ตะวันตก','ตะวันตก'],
             'การเมือง': ['กลาง','กลาง'],
             'อุตุนิยมวิทยา': ['กลาง','กลาง'],
             'การท่องเที่ยว': ['กลาง','กลาง']
}
region10 = { 'จังหวัด': ['เพชรบุรี', 'ประจวบคีรีขันธ์'],
             'ภูมิศาสตร์': ['ตะวันตก','ตะวันตก'],
             'การเมือง': ['กลาง','กลาง'],
             'อุตุนิยมวิทยา': ['ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย','ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย'],
             'การท่องเที่ยว': ['กลาง','กลาง']
}
region11 = { 'จังหวัด': ['ชุมพร', 'นครศรีธรรมราช','นราธิวาส', 'ปัตตานี', 'พัทลุง', 'สงขลา', 'สุราษฎร์ธานี', 'ยะลา'],
             'ภูมิศาสตร์': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้'],
             'การเมือง': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้'],
             'อุตุนิยมวิทยา': ['ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย', 'ใต้ฝั่งตะวันออก/ใต้ฝั่งอ่าวไทย'],
             'การท่องเที่ยว': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้']
}
region12 = { 'จังหวัด': ['กระบี่', 'พังงา', 'ภูเก็ต', 'ระนอง', 'สตูล', 'ตรัง'],
             'ภูมิศาสตร์': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้'],
             'การเมือง': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้'],
             'อุตุนิยมวิทยา': ['ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน', 'ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน', 'ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน', 'ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน', 'ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน', 'ใต้ฝั่งตะวันตก/ใต้ฝั่งอันดามัน'],
             'การท่องเที่ยว': ['ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้', 'ใต้']
}

In [ ]:
df_region1 = pd.DataFrame.from_dict(region1)
df_region2 = pd.DataFrame.from_dict(region2)
df_region3 = pd.DataFrame.from_dict(region3)
df_region4 = pd.DataFrame.from_dict(region4)
df_region5 = pd.DataFrame.from_dict(region5)
df_region6 = pd.DataFrame.from_dict(region6)
df_region7 = pd.DataFrame.from_dict(region7)
df_region8 = pd.DataFrame.from_dict(region8)
df_region9 = pd.DataFrame.from_dict(region9)
df_region10 = pd.DataFrame.from_dict(region10)
df_region11 = pd.DataFrame.from_dict(region11)
df_region12 = pd.DataFrame.from_dict(region12)

In [ ]:
region_concat = pd.concat([df_region1, df_region2, df_region3, df_region4, df_region5, df_region6,
                            df_region7, df_region8, df_region9, df_region10, df_region11, df_region12])  
print(region_concat.shape)
region_concat.head()

(77, 5)


จังหวัด          ภูมิศาสตร์            การเมือง       อุตุนิยมวิทยา  \
0  อำนาจเจริญ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
1      บึงกาฬ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
2   บุรีรัมย์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
3     ชัยภูมิ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
4   กาฬสินธุ์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   

        การท่องเที่ยว  
0  ตะวันออกเฉียงเหนือ  
1  ตะวันออกเฉียงเหนือ  
2  ตะวันออกเฉียงเหนือ  
3  ตะวันออกเฉียงเหนือ  
4  ตะวันออกเฉียงเหนือ

In [ ]:
region_concat.isnull().sum()

จังหวัด          0
ภูมิศาสตร์       0
การเมือง         0
อุตุนิยมวิทยา    0
การท่องเที่ยว    0
dtype: int64

In [ ]:
##########################################################################################################################################

In [ ]:
print(region_concat.shape)              ### ตารางพร้อมใช้งาน ของ region
region_concat.head()

(77, 5)


จังหวัด          ภูมิศาสตร์            การเมือง       อุตุนิยมวิทยา  \
0  อำนาจเจริญ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
1      บึงกาฬ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
2   บุรีรัมย์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
3     ชัยภูมิ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
4   กาฬสินธุ์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   

        การท่องเที่ยว  
0  ตะวันออกเฉียงเหนือ  
1  ตะวันออกเฉียงเหนือ  
2  ตะวันออกเฉียงเหนือ  
3  ตะวันออกเฉียงเหนือ  
4  ตะวันออกเฉียงเหนือ

# 3.Elderly

ref: https://www.dop.go.th/th/know/side/1/1/1099

- ข้อมูลผู้สูงอายุของประเทศไทย ที่มีอายุ 60 ขึ้นไป

In [ ]:
elderly = pd.read_excel('จำนวนผู้สูงอายุที่มีอายุมากว่า 60 ปี 2564.xlsx')
print(elderly.shape)
elderly.head()

(77, 6)


ลำดับ  รหัส      ชื่อจังหวัด       ชาย      หญิง        รวม
0    1.0  10.0    กรุงเทพมหานคร  477117.0  675178.0  1152295.0
1    2.0  11.0      สมุทรปราการ   94967.0  129053.0   224020.0
2    3.0  12.0          นนทบุรี  109375.0  148338.0   257713.0
3    4.0  13.0         ปทุมธานี   79657.0  105714.0   185376.0
4    5.0  14.0  พระนครศรีอยุธยา   68172.0   92624.0   160796.0

In [ ]:
elderly = elderly[['ชื่อจังหวัด','ชาย','หญิง','รวม']].rename(columns={'ชื่อจังหวัด':'province',
                                                                    'ชาย':'male',
                                                                    'หญิง':'female',
                                                                    'รวม':'total'})
print(elderly.shape)
elderly.head()

(77, 4)


province      male    female      total
0    กรุงเทพมหานคร  477117.0  675178.0  1152295.0
1      สมุทรปราการ   94967.0  129053.0   224020.0
2          นนทบุรี  109375.0  148338.0   257713.0
3         ปทุมธานี   79657.0  105714.0   185376.0
4  พระนครศรีอยุธยา   68172.0   92624.0   160796.0

In [ ]:
elderly.isnull().sum()      # เช็คค่าว่าง

province    0
male        0
female      0
total       0
dtype: int64

In [ ]:
########################################################################################################################

In [ ]:
print(elderly.shape)    #### ตารางพร้อมงานของ elderly
elderly.head()

(77, 4)


province      male    female      total
0    กรุงเทพมหานคร  477117.0  675178.0  1152295.0
1      สมุทรปราการ   94967.0  129053.0   224020.0
2          นนทบุรี  109375.0  148338.0   257713.0
3         ปทุมธานี   79657.0  105714.0   185376.0
4  พระนครศรีอยุธยา   68172.0   92624.0   160796.0

# รวมตาราง

In [ ]:
print(df_drop.shape)    ############## ตารางสุดท้าย พร้อมใช้งานของ covid_data
df_drop.head(3)

(1734198, 7)


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset  
0     กรุงเทพมหานคร     กรุงเทพมหานคร  
1     กรุงเทพมหานคร     กรุงเทพมหานคร  
2            นครปฐม             เมือง

In [ ]:
print(region_concat.shape)              ### ตารางพร้อมใช้งาน ของ region
region_concat.head(3)

(77, 5)


จังหวัด          ภูมิศาสตร์            การเมือง       อุตุนิยมวิทยา  \
0  อำนาจเจริญ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
1      บึงกาฬ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   
2   บุรีรัมย์  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ  ตะวันออกเฉียงเหนือ   

        การท่องเที่ยว  
0  ตะวันออกเฉียงเหนือ  
1  ตะวันออกเฉียงเหนือ  
2  ตะวันออกเฉียงเหนือ

In [ ]:
print(elderly.shape)    #### ตารางพร้อมงานของ elderly
elderly.head(3)

(77, 4)


province      male    female      total
0  กรุงเทพมหานคร  477117.0  675178.0  1152295.0
1    สมุทรปราการ   94967.0  129053.0   224020.0
2        นนทบุรี  109375.0  148338.0   257713.0

In [ ]:
covid_region = df_drop.merge(region_concat,how='left',left_on='province_of_onset',right_on='จังหวัด')     # ต่อตาราง covid ด้วยตาราง region
print(covid_region.shape)
covid_region.head(3)

(1734198, 12)


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset        จังหวัด ภูมิศาสตร์ การเมือง  \
0     กรุงเทพมหานคร     กรุงเทพมหานคร  กรุงเทพมหานคร       กลาง     กลาง   
1     กรุงเทพมหานคร     กรุงเทพมหานคร  กรุงเทพมหานคร       กลาง     กลาง   
2            นครปฐม             เมือง         นครปฐม       กลาง     กลาง   

  อุตุนิยมวิทยา การท่องเที่ยว  
0          กลาง          กลาง  
1          กลาง          กลาง  
2          กลาง          กลาง

In [ ]:
all_data = covid_region.merge(elderly,how='left',left_on='province_of_onset',right_on='province')     # ต่อตาราง covid ด้วยตาราง region
print(all_data.shape)
all_data.head(3)

(1734198, 16)


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset        จังหวัด ภูมิศาสตร์ การเมือง  \
0     กรุงเทพมหานคร     กรุงเทพมหานคร  กรุงเทพมหานคร       กลาง     กลาง   
1     กรุงเทพมหานคร     กรุงเทพมหานคร  กรุงเทพมหานคร       กลาง     กลาง   
2            นครปฐม             เมือง         นครปฐม       กลาง     กลาง   

  อุตุนิยมวิทยา การท่องเที่ยว       province      male    female      total  
0          กลาง          กลาง  กรุงเทพมหานคร  477117.0  675178.0  1152295.0  
1          กลาง          กลาง  กรุงเทพมหานคร  477117.0  675178.0  1152295.0  
2          กลาง          กลาง         นครปฐม   72486.0   99258.0   171744.0

In [ ]:
all_data = all_data.drop(columns=['จังหวัด','province'])
print(all_data.shape)
all_data.head(3)

(1734198, 14)


No. announce_date   sex   age                              risk  \
0    1    2020-01-12  หญิง  61.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง  74.0  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง  73.0  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา  \
0     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
1     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
2            นครปฐม             เมือง       กลาง     กลาง          กลาง   

  การท่องเที่ยว      male    female      total  
0          กลาง  477117.0  675178.0  1152295.0  
1          กลาง  477117.0  675178.0  1152295.0  
2          กลาง   72486.0   99258.0   171744.0

In [ ]:
all_data.isnull().any()

No.                  False
announce_date        False
sex                  False
age                  False
risk                 False
province_of_onset    False
district_of_onset    False
ภูมิศาสตร์           False
การเมือง             False
อุตุนิยมวิทยา        False
การท่องเที่ยว        False
male                 False
female               False
total                False
dtype: bool

#### เปลี่ยนชนิดของข้อมูล

In [ ]:
all_data.dtypes

No.                           int64
announce_date        datetime64[ns]
sex                          object
age                         float64
risk                         object
province_of_onset            object
district_of_onset            object
ภูมิศาสตร์                   object
การเมือง                     object
อุตุนิยมวิทยา                object
การท่องเที่ยว                object
male                          int64
female                        int64
total                         int64
dtype: object

In [ ]:
all_data = all_data.astype({'age':'int','male':'int','female':'int','total':'int'})     # เปลี่ยนชนิดข้อมูลเป็น int
print(all_data.dtypes)
all_data.head(3)

No.                           int64
announce_date        datetime64[ns]
sex                          object
age                           int64
risk                         object
province_of_onset            object
district_of_onset            object
ภูมิศาสตร์                   object
การเมือง                     object
อุตุนิยมวิทยา                object
การท่องเที่ยว                object
male                          int64
female                        int64
total                         int64
dtype: object


No. announce_date   sex  age                              risk  \
0    1    2020-01-12  หญิง   61  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง   74  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง   73  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา  \
0     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
1     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
2            นครปฐม             เมือง       กลาง     กลาง          กลาง   

  การท่องเที่ยว    male  female    total  
0          กลาง  477117  675178  1152295  
1          กลาง  477117  675178  1152295  
2          กลาง   72486   99258   171744

### แยกข้อมูล

- เนื่องจากขนาดไฟล์ใหญ่เกินไปจึงแยกข้อมูลออกเป็นส่วนๆ

In [ ]:
covid_1 = all_data[all_data['announce_date'] < '2021-08-01']     # เลือกข้อมูลตั้งแต่วันแรก ถึง 2021-07-31      # หรือ ตั้งแต่วันแรกจนถึงเดือน 7 ปี 2021
print(covid_1.shape)                                            
covid_1.head(3)

(289132, 14)


No. announce_date   sex  age                              risk  \
0    1    2020-01-12  หญิง   61  คนต่างชาติเดินทางมาจากต่างประเทศ   
1    2    2020-01-17  หญิง   74  คนต่างชาติเดินทางมาจากต่างประเทศ   
2    3    2020-01-22  หญิง   73  คนต่างชาติเดินทางมาจากต่างประเทศ   

  province_of_onset district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา  \
0     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
1     กรุงเทพมหานคร     กรุงเทพมหานคร       กลาง     กลาง          กลาง   
2            นครปฐม             เมือง       กลาง     กลาง          กลาง   

  การท่องเที่ยว    male  female    total  
0          กลาง  477117  675178  1152295  
1          กลาง  477117  675178  1152295  
2          กลาง   72486   99258   171744

In [ ]:
covid_2 = all_data[(all_data['announce_date'] >= '2021-08-01') & (all_data['announce_date'] < '2021-09-01')]     # เลือกข้อมูล เดือน 8 ปี 2021
print(covid_2.shape)
covid_2.head(3)

(297438, 14)


No. announce_date   sex  age   risk province_of_onset  \
289132  597493    2021-08-01  หญิง   34  อื่นๆ     กรุงเทพมหานคร   
289133  597494    2021-08-01   ชาย   50  อื่นๆ     กรุงเทพมหานคร   
289134  597495    2021-08-01  หญิง    7  อื่นๆ     กรุงเทพมหานคร   

       district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
289132     กรุงเทพมหานคร       กลาง     กลาง          กลาง          กลาง   
289133     กรุงเทพมหานคร       กลาง     กลาง          กลาง          กลาง   
289134     กรุงเทพมหานคร       กลาง     กลาง          กลาง          กลาง   

          male  female    total  
289132  477117  675178  1152295  
289133  477117  675178  1152295  
289134  477117  675178  1152295

In [ ]:
covid_3 = all_data[(all_data['announce_date'] >= '2021-09-01') & (all_data['announce_date'] < '2021-10-01')]     # เลือกข้อมูล เดือน 9 ปี 2021
print(covid_3.shape)
covid_3.head(3)

(198692, 14)


No. announce_date   sex  age   risk province_of_onset  \
586570  1204730    2021-09-01  หญิง   49  อื่นๆ         เชียงใหม่   
586571  1204731    2021-09-01   ชาย   29  อื่นๆ         เชียงใหม่   
586572  1204732    2021-09-01  หญิง   16  อื่นๆ         เชียงใหม่   

       district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
586570          สันกำแพง      เหนือ    เหนือ         เหนือ         เหนือ   
586571             เมือง      เหนือ    เหนือ         เหนือ         เหนือ   
586572             สารภี      เหนือ    เหนือ         เหนือ         เหนือ   

          male  female   total  
586570  162707  202855  365562  
586571  162707  202855  365562  
586572  162707  202855  365562

In [ ]:
covid_4 = all_data[(all_data['announce_date'] >= '2021-10-01') & (all_data['announce_date'] < '2021-12-01')]     # เลือกข้อมูล เดือน 10,11 ปี 2021
print(covid_4.shape)
covid_4.head(3)

(316660, 14)


No. announce_date   sex  age  \
785262  1603476    2021-10-01  หญิง   30   
785263  1603477    2021-10-01  หญิง   32   
785264  1603478    2021-10-01  หญิง   38   

                                               risk province_of_onset  \
785262                                        อื่นๆ             พะเยา   
785263                                        อื่นๆ             พะเยา   
785264  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          เชียงราย   

       district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
785262           เชียงคำ      เหนือ    เหนือ         เหนือ         เหนือ   
785263           เชียงคำ      เหนือ    เหนือ         เหนือ         เหนือ   
785264             เมือง      เหนือ    เหนือ         เหนือ         เหนือ   

          male  female   total  
785262   50355   58897  109252  
785263   50355   58897  109252  
785264  115166  134523  249689

In [ ]:
covid_5 = all_data[(all_data['announce_date'] >= '2021-12-01') & (all_data['announce_date'] < '2022-02-01')]     # เลือกข้อมูล เดือน 11 ปี 2021 - เดือน 1 ปี 2022
print(covid_5.shape)
covid_5.head(3)

(166203, 14)


No. announce_date   sex  age  \
1101922  2115873    2021-12-01  หญิง   63   
1101923  2115874    2021-12-01  หญิง   40   
1101924  2115875    2021-12-01   ชาย   38   

                                                risk province_of_onset  \
1101922  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้          เชียงราย   
1101923                                        อื่นๆ          เชียงราย   
1101924                                        อื่นๆ          เชียงราย   

        district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
1101922              เทิง      เหนือ    เหนือ         เหนือ         เหนือ   
1101923             เมือง      เหนือ    เหนือ         เหนือ         เหนือ   
1101924              เทิง      เหนือ    เหนือ         เหนือ         เหนือ   

           male  female   total  
1101922  115166  134523  249689  
1101923  115166  134523  249689  
1101924  115166  134523  249689

In [ ]:
covid_6 = all_data[(all_data['announce_date'] >= '2022-02-01') & (all_data['announce_date'] < '2022-03-01')]     # เลือกข้อมูล เดือน 2 ปี 2022
print(covid_6.shape)                                            
covid_6.head(3)

(228602, 14)


No. announce_date   sex  age  \
1268125  2440543    2022-02-01  หญิง   35   
1268126  2440544    2022-02-01  หญิง   19   
1268127  2440547    2022-02-01   ชาย   44   

                                                risk province_of_onset  \
1268125                                        อื่นๆ             พะเยา   
1268126  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   
1268127  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้             พะเยา   

        district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
1268125           เชียงคำ      เหนือ    เหนือ         เหนือ         เหนือ   
1268126           เชียงคำ      เหนือ    เหนือ         เหนือ         เหนือ   
1268127          ดอกคำใต้      เหนือ    เหนือ         เหนือ         เหนือ   

          male  female   total  
1268125  50355   58897  109252  
1268126  50355   58897  109252  
1268127  50355   58897  109252

In [ ]:
covid_7 = all_data[all_data['announce_date'] >= '2022-03-01']      # เลือกข้อมูลตั้งแต่ เดือน 3 ปี 2022 - ข้อมูลล่าสุดที่โหลดมา
print(covid_7.shape)                                            
covid_7.head(3)

(237471, 14)


No. announce_date   sex  age  \
1496727  2891928    2022-03-01   ชาย   35   
1496728  2891929    2022-03-01  หญิง   35   
1496729  2891930    2022-03-01  หญิง   35   

                                                risk province_of_onset  \
1496727                                        อื่นๆ        แม่ฮ่องสอน   
1496728  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้        แม่ฮ่องสอน   
1496729  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้        แม่ฮ่องสอน   

        district_of_onset ภูมิศาสตร์ การเมือง อุตุนิยมวิทยา การท่องเที่ยว  \
1496727             เมือง      เหนือ    เหนือ         เหนือ         เหนือ   
1496728               ปาย      เหนือ    เหนือ         เหนือ         เหนือ   
1496729        แม่สะเรียง      เหนือ    เหนือ         เหนือ         เหนือ   

          male  female  total  
1496727  17515   18544  36059  
1496728  17515   18544  36059  
1496729  17515   18544  36059

## โหลดข้อมูล

In [ ]:
#covid_1.to_csv('covid_frist to 07-2021.csv',index=False)      # โหลดตาราง covid ตั้งแต่วันแรกจนถึงเดือน 7 ปี 2021
#covid_2.to_csv('covid_08-2021.csv',index=False)      # โหลดตาราง covid เดือน 8 ปี 2021
#covid_3.to_csv('covid_09-2021.csv',index=False)      # โหลดตาราง covid เดือน 9 ปี 2021
#covid_4.to_csv('covid_10 and 11-2021.csv',index=False)      # โหลดตาราง covid เดือน 10,11 ปี 2021
#covid_5.to_csv('covid_11-2021 to 1-2022.csv',index=False)      # โหลดตาราง covid เดือน 11 ปี 2021 - เดือน 1 ปี 2022
#covid_6.to_csv('covid_02-2022.csv',index=False)      # โหลดตาราง covid เดือน 2 ปี 2022
#covid_7.to_csv('covid_03-2022 to last.csv',index=False)      # โหลดตาราง covid เดือน 2 ปี 2022 - ข้อมูลล่าสุดที่โหลดมา